libyul/optimiser/Semantics.h

```diff
@@ -69,6 +107,7 @@ class SideEffectsCollector: public ASTWalker
 	bool invalidatesStorage() const { return m_sideEffects.storage == SideEffects::Write; }
 	bool invalidatesMemory() const { return m_sideEffects.memory == SideEffects::Write; }
 
+	SideEffects sideEffects() { return m_sideEffects; }
```

Comment:
This returns the SideEffects directly and thus does not apply the `cannotLoop`-logic. I really think we should just set movable to false if functions contain loops.